### ISSUE TITLE
Bug: Autoregressive training loop is incomplete, connection loss is not calculated

### ISSUE THREAD

**Issue by timlawrenz** [timlawrenz/jubilant-palm-tree#132]
### Description

The `train_autoregressive.py` script is currently incomplete, leading to a poorly trained autoregressive decoder that cannot generate meaningful code. The core of the issue is that the training loop only calculates the loss for node type prediction and explicitly skips the loss calculation for the connections between nodes.

As a result, the model learns *what* nodes to generate but has no information about *how* they should be connected to form a valid Abstract Syntax Tree. This is the primary reason why the generated code is a simple, flat structure.

### Location of the Bug

The bug is in the `train_step` function within `train_autoregressive.py`:

```python
# ...
# Calculate loss for this step
node_type_loss = F.cross_entropy(node_type_logits, target_tensor)

# For now, skip connection loss as it requires more complex target preparation
# In a full implementation, you'd calculate connection loss here too
step_loss = node_type_loss 
# ...
```

### Task

Your task is to fix the training loop to correctly calculate and incorporate the connection loss.

1.  **Prepare Target Connection Data**:
    *   Ensure that the `create_autoregressive_data_loader` and the `AutoregressiveASTDataset` (in `src/data_processing.py`) provide the target connection information for each training step. This should be a tensor representing the correct connections for the node being predicted.

2.  **Update the `train_step` Function**:
    *   Remove the comment and the line that skips the connection loss.
    *   Calculate the `connection_loss` using `F.binary_cross_entropy` between the model's `connection_probs` output and the target connection tensor from the data loader.
    *   Combine the losses: `step_loss = node_type_loss + connection_loss`.

3.  **Update the `validate_step` Function**:
    *   Apply the same fix to the validation step to ensure that validation loss is also calculated correctly.

### Acceptance Criteria

*   The `train_step` and `validate_step` functions in `train_autoregressive.py` calculate both `node_type_loss` and `connection_loss`.
*   The total `step_loss` is a combination of both losses.
*   The model, when retrained with this corrected script, should be able to generate more complex and properly structured code, as it will have learned from the AST's edge structure.

This is a critical bug fix required to make the Phase 7 autoregressive decoder functional.
